### PR TITLE
docs: readme for proto files

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

Adds readme for proto files regarding the Conduit API.

### CODE REVIEW

File: proto/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Because we use remote code generation provided by the Buf schema registry there
+is no locally generated code. When developing locally we don't want to push a
+new version of the proto files every time we make a change, that's why in that
+case we can switch to locally generated protobuf code.
+
+To switch to locally generated protobuf code follow the following steps:
+- run `cd proto && buf generate`
+- cd into the newly generated folder `proto/gen`
+- create a `go.mod` file by running `go mod init go.buf.build/conduitio/conduit/conduitio/conduit && go mod tidy`
+- cd into the root of the project and run `go mod edit -replace go.buf.build/conduitio/conduit/conduitio/conduit=./proto/gen && go mod tidy`
```
Comment (maha-hajja) [inline proto/README.md:32]: should we add something about how to reverse this command? after the local development is done?

Comment (hariso) [inline proto/README.md:32]: Nitpick: would be cool if we had this as a make target.

**Review by hariso — APPROVED** [review]

File: proto/README.md (hunk 1 above)
Comment (lovromazgon) [reply proto/README.md:32]: Agreed, I hope we can live with this piece of documentation for now though 🙃

Comment (hariso) [reply proto/README.md:32]: We definitely can.:)

**Pull request merged**